docs/guides/wireguard/client.md

```diff
@@ -0,0 +1,208 @@
+
+<!-- markdownlint-disable code-block-style -->
+??? info "All commands described below at once"
+    ```bash
+    sudo -i
+    cd /etc/wireguard
+    umask 077
+
+    name="client_name"
+
+    wg genkey | tee "${name}.key" | wg pubkey > "${name}.pub"
+    wg genpsk > "${name}.psk"
+
+    echo "[Peer]" >> /etc/wireguard/wg0.conf
+    echo "PublicKey = $(cat "${name}.pub")" >> /etc/wireguard/wg0.conf
+    echo "PresharedKey = $(cat "${name}.psk")" >> /etc/wireguard/wg0.conf
+    echo "AllowedIPs = 10.100.0.2/32, fd08:4711::2/128" >> /etc/wireguard/wg0.conf
+
+    systemctl restart wg-quick@wg0
+
+    echo "[Interface]" > "${name}.conf"
+    echo "Address = 10.100.0.2/32, fd08:4711::2/128" >> "${name}.conf" # May need editing
+    echo "DNS = 10.100.0.1" >> "${name}.conf"                          # Your Pi-hole's IP
+    echo "PrivateKey = $(cat "${name}.key")" >> "${name}.conf"
+    echo "PublicKey = $(cat server.pub)" >> "${name}.conf"
+    echo "PresharedKey = $(cat "${name}.psk")" >> "${name}.conf"
+
+    qrencode -t ansiutf8 -r "${name}.conf"
+
+    exit
+    ```
+<!-- markdownlint-disable code-block-style -->
+
+## Key generation
+
+We generate a key-pair for the client `NAME` (replace accordingly everywhere below):
+
+```bash
+sudo -i
+cd /etc/wireguard
+umask 077
+name="client_name"
+wg genkey | tee "${name}.key" | wg pubkey > "${name}.pub"
+```
+
+## PSK Key generation
+
+We furthermore recommend generating a pre-shared key (PSK) in addition to the keys above. This adds an additional layer of symmetric-key cryptography to be mixed into the already existing public-key cryptography and is mainly for post-quantum resistance. A pre-shared key should be generated for each peer pair and *should not be reused*.
+
+```bash
+wg genpsk > "${name}.psk"
+```
+
+## Add client to server configuration
+
+Add the new client by running the command:
+
+```bash
+echo "[Peer]" >> /etc/wireguard/wg0.conf
+echo "PublicKey = $(cat "${name}.pub")" >> /etc/wireguard/wg0.conf
+echo "PresharedKey = $(cat "${name}.psk")" >> /etc/wireguard/wg0.conf
+echo "AllowedIPs = 10.100.0.2/32, fd08:4711::2/128" >> /etc/wireguard/wg0.conf
+```
+
+<!-- markdownlint-disable code-block-style -->
+!!! info "Client IP address"
+    Make sure to increment the IP address for any further client! We add the first client with the IP addresses `10.100.0.2` and `fd08:4711::2/128` in this example (`10.100.0.1` and `fd08:4711::1/128` are the server)
+<!-- markdownlint-disable code-block-style -->
+
+Restart your server to load the new client config:
+
+```bash
+systemctl restart wg-quick@wg0
+```
+
+After a restart, the server file should look like:
+
+```toml
```

Comment:
Are you sure `toml` is right? I see the GitHub parser produces wrong output.

Reply:
`toml` looks odd to me actually and I don't think there is a `toml` format? Could do with nothing and just leave it unstyled. 

http://www.rubycoloredglasses.com/2013/04/languages-supported-by-github-flavored-markdown/ shows the languages, but I don't think we're limited to GFM, we could use any lang spec supported by plugin.

Reply:
`bash` displays a nice output. Comments are noted and everything else stays plain. (Except that bool)
![Screenshot 2020-10-18 224323](https://user-images.githubusercontent.com/2782850/96406053-740ff680-1193-11eb-9b98-1620d6bc41c3.png)

Reply:
Yes, it's true that GitHub is producing incorrect output, however, this shouldn't stop us from using a formatter which is supported by what matters in the end - `mkdocs`. I'm fine with plain if you prefer this, but still think color is nice for configuration files on long pages.

If you could point me to the source code of the highlighter used in `mkdocs`, I can make a PR there to add `wireguard.conf` as new format with 100% proper highlighting

Reply:
What you see in the netlify preview link is what you'll see on the actual deployed site.

Reply:
I'd go with whatever produces better output. Toml does not seem to be close to what we want, so maybe what @dschaper suggested is better compared to no highlighting?

It's a tricky situation for sure, hence why, personally, I go with plan text when there isn't a 100% language I can use.